bowling-game/tabcorp/31-05-2017/test.js

```diff
@@ -0,0 +1,53 @@
+const assert = require('assert');
+
+describe('Bowling Game', () => {
+
+  const createGame = () => {
+    let rollsResult = []
+    return {
+      roll: function(pinsTakenDown) {
+        rollsResult.push(pinsTakenDown)
+      },
+      score: function() {
+        const pinsTakenDownToRolls = function(pinsTakenDown) {
+          const roll = {
+            pinsTakenDown: pinsTakenDown,
+            strike: false
+          }
+          if (pinsTakenDown === 10) {
+            roll.strike = true
+          }
+          return roll;
+        }
+        const toSumOfPinsTakenDown = (previousSum, currentRoll) => {
+          return previousSum + currentRoll.pinsTakenDown;
+        }
+
+        const rolls = rollsResult.map(pinsTakenDownToRolls)
+        return rolls.reduce(toSumOfPinsTakenDown, 0);
```

Comment:
Here’s my 2 cents: Although I can’t point the difference in the code because it’s quite small, afaik on cqRs you would have different “paths” for query and for command. It would essentially have different models for query and command, maybe having separated databases (optimised for read and write) and even moving to two separate systems where each can scale independently.